refactor SQL database tests

Ladicek opened the pull request:
Previously, the SQL database was deployed _after_ the testing
application, which is wrong. It used to work, but it no longer
does, so it's time to fix it.

This commit introduces a special Arquillian extension that
observes the `BeforeClass` and `AfterClass` events (that is,
events fired by Arquillian just before/after tests from a single
class are executed) and deploy/undeploy the database at that
point in time.

That requires switching off automatic application deployment
during Arquillian Cube initialization. Instead, application
is deployed using the `@OpenShiftResource` annotation. That
annotation is also implemented using a `Before/AfterClass`
observer, so it's important to make sure that the database
deployment observer runs sooner.

A little bit of code cleanup elsewhere is also included.

Ladicek committed:
refactor SQL database tests

Previously, the SQL database was deployed _after_ the testing
application, which is wrong. It used to work, but it no longer
does, so it's time to fix it.

This commit introduces a special Arquillian extension that
observes the `BeforeClass` and `AfterClass` events (that is,
events fired by Arquillian just before/after tests from a single
class are executed) and deploy/undeploy the database at that
point in time.

That requires switching off automatic application deployment
during Arquillian Cube initialization. Instead, application
is deployed using the `@OpenShiftResource` annotation. That
annotation is also implemented using a `Before/AfterClass`
observer, so it's important to make sure that the database
deployment observer runs sooner.

A little bit of code cleanup elsewhere is also included.

Ladicek commented:
@juangon Could you test if this works for you?

Diffs are probably bad for review, better check it out locally and go through it in an IDE.

juangon commented:
@Ladicek sure! Yes I need some time to understand those Arquillian extensions. Will check in parallel of executing tests...

Ladicek commented:
Thanks!

Ladicek committed:
cleanup project before test class

This commits adds an annotation `@ProjectCleanup` which, when applied
to a test class, makes sure that the current OpenShift project
is cleaned up before the test class is executed.

All the SQL database tests use this annotation now.

Ladicek commented:
Added a commit for cleaning up the OpenShift project before each test.

Ladicek commented:
Thanks
